src/modules/QtQuick/BorderImage.js

```diff
@@ -3,31 +3,28 @@ registerQmlType({
   name:     'BorderImage',
   versions: /.*/,
   baseClass: 'Item',
-  constructor: function QMLBorderImage(meta) {
+  enums: {
+    BorderImage: {
+      Stretch: 'stretch', Repeat: 'repeat', Round: 'round',
+      Null: 1, Ready: 2, Loading: 3, Error: 4
+    }
+  },
+  properties: {
+    source: 'url',
+    horizontalTileMode: { type: 'enum', initialValue: 'stretch' }, // BorderImage.Stretch
+    verticalTileMode: { type: 'enum', initialValue: 'stretch' }, // BorderImage.Stretch
+    status: { type: 'enum', initialValue: 1 } // BorderImage.Null
+  }
+}, class {
+  constructor(meta) {
     callSuper(this, meta);
     var self = this;
 
-    this.BorderImage = {
-        // tileMode
-        Stretch: "stretch",
-        Repeat: "repeat",
-        Round: "round",
-        // status
-        Null: 1,
-        Ready: 2,
-        Loading: 3,
-        Error: 4
-    }
-
-    createProperty("url", this, "source");
-    createProperty("enum", this, "status", {initialValue: this.BorderImage.Null});
     this.border = new QObject(this);
     createProperty("int", this.border, "left");
     createProperty("int", this.border, "right");
     createProperty("int", this.border, "top");
     createProperty("int", this.border, "bottom");
```

Comment:
We should find a declarative solution for these as well, i.e. registering a type QuickBorder (or similar), which would be used here. But that can be done in another PR.